tools/src/main/scala/caliban/tools/compiletime/CompileTime.scala

```diff
@@ -0,0 +1,78 @@
+package caliban.tools.compiletime
+
+import caliban.GraphQL
+import caliban.tools.Codegen.GenType
+import caliban.tools.{ CalibanCommonSettings, Codegen, SchemaLoader }
+import zio.blocking.effectBlocking
+import zio.{ ExitCode, URIO, ZEnv }
+
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+
+object CompileTime {
+  import Utils._
+
+  final case class GenerateClientsSettings(
+    clientName: String,
+    packageName: String,
+    scalafmtPath: Option[String] = None,
+    headers: List[(String, String)] = List.empty,
+    genView: Boolean = false,
+    scalarMappings: List[(String, String)] = List.empty,
+    imports: List[String] = List.empty,
+    splitFiles: Boolean = false,
+    enableFmt: Boolean = true,
+    extensibleEnums: Boolean = false
+  )                              {
+    def toCalibanCommonSettings: CalibanCommonSettings =
+      CalibanCommonSettings(
+        clientName = Some(clientName),
+        scalafmtPath = scalafmtPath,
+        headers = headers,
+        packageName = Some(packageName),
+        genView = Some(genView),
+        scalarMappings = scalarMappings,
+        imports = imports,
+        splitFiles = Some(splitFiles),
+        enableFmt = Some(enableFmt),
+        extensibleEnums = Some(extensibleEnums)
+      )
+  }
+  object GenerateClientsSettings {
+    def default: GenerateClientsSettings = GenerateClientsSettings(clientName = "Client", packageName = "generated")
+  }
+
+  def generateClient[R](args: List[String])(settings: GenerateClientsSettings, api: GraphQL[R]): URIO[ZEnv, ExitCode] =
+    args match {
+      case baseDir :: _ =>
+        (for {
+          toPath <- effectBlocking {
+                      val directory = s"$baseDir/src/main/scala/${packagePath(settings.packageName)}"
+
+                      createDirectories(directory)
+
+                      s"$directory/${settings.clientName}.scala"
```

Comment:
This looks like it doesn't support `splitFiles`, do you need help with that?

Reply:
True. I don't know what `splitFiles` does. If you have an idea on how to fix this problem, go for it. Otherwise, I'll try to propose a solution.

Reply:
It generates the code is multiple files instead of a big one.

Reply:
In the new version, the `splitFiles` works fine. I'll write some test about it.